Sources/AEPOptimize/OptimizeError.swift

```diff
@@ -0,0 +1,81 @@
+// Delete this line
+/*
+ Copyright 2024 Adobe. All rights reserved.
+ This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License. You may obtain a copy
+ of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software distributed under
+ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ OF ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ */
+
+import AEPCore
+import Foundation
+
+/// AEPOptimizeError class used to create AEPOptimizeError from error details received from Experience Edge.
+@objc(AEPOptimizeError)
+public class AEPOptimizeError: NSObject, Error {
+    typealias HTTPResponseCodes = OptimizeConstants.HTTPResponseCodes
+    public let type: String?
```

Comment:
is this type property being used?

Reply:
type property is received in edge error response content and we have kept it for that purpose. For timeout or any other invalid request case on optimize SDK side this will be nil.